conans/client/cmd/profile.py

```diff
@@ -21,13 +21,23 @@ def _get_profile_keys(key):
 
 
 def cmd_profile_list(cache_profiles_path, output):
-    folder = cache_profiles_path
-    if os.path.exists(folder):
-        return [name for name in os.listdir(folder)
-                if not os.path.isdir(os.path.join(folder, name))]
-    else:
+    profiles = []
+    if os.path.exists(cache_profiles_path):
+        for current_directory, dirs, files in os.walk(cache_profiles_path):
+            # We don't want to list files from hidden directory
+            dirs[:] = [d for d in dirs if not d.startswith('.')]
+
+            for filename in files:
```

Comment:
Do we want to ignore hidden profile files?

```python
for filename in files:
    if filename.startswith('.'):
        continue
    ...
```

Reply:
Yeah we might want to ignore hidden profile file. I didn't add this check because it could potentially change behavior for current user i believe. But it might indeed make more sense to not list hidden files.

Reply:
Given your question about linked folders,  I imagine that right now one can use a profile that starts with `.` (even one inside a _hidden_ folder), so should we avoid listing them? If you can use it, you should be able to see it in the list...

I would never use _hidden files_ or _hidden folders_ to store my profiles, but I think there is no reason to forbid them, so maybe we should show/list them too. I'll ping @memsharded for an opinion on this.

Reply:
I would simplify, symlinks and hidden folders in the profile folder sounds a bit crazy. Let's try to keep it simple.

Reply:
Yes, I agree with you, @lasote, but now you can use (`conan profile show XXX` or `conan .... --profile=XXX`) hidden files (and files inside _hidden folders_) as profiles, so it makes sense to list them too using `conan profile list`.